apps/sample-app/src/components/utils.tsx

```diff
@@ -81,12 +82,16 @@ export const handleMail = async (a: AvatarItem): Promise<void> => {
 };
 
 export const handleOpenMailItem = async (emailItem: MessageListItem): Promise<void> => {
-  if (!mail.isSupported()) {
-    alert('open mail item is not supported');
+  const openMailParams: mail.OpenMailItemParams = {
+    itemId: emailItem.key || '',
+  };
+  await mail.openMailItem(openMailParams);
+};
+
+export const shouldShowMeeting = (meeting: Message): boolean => {
+  if (meeting['showAs'] === 'free' || meeting['isCancelled']) {
+    return false;
   } else {
-    const openMailParams: mail.OpenMailItemParams = {
-      itemId: emailItem.key || '',
-    };
-    await mail.openMailItem(openMailParams);
+    return true;
   }
```

Comment:
As someone who has an unreasonable obsession with writing very compact but still readable code, you _could_ consider writing it like this:

```
return (meeting['showAs'] !== 'free' && !meeting['isCancelled'])
```